<?xml version="1.0"?>
<!--
MIT License

Copyright (c) 2022 Gotelli Andrea

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
-->


<launch>

    <arg name="number_of_limbs" value="6" />
    <arg name="robot_name" value="Stewart_Gough_6EFR" />

    <param name="limb_name_prefix" type="string" value="limb" />
    <param name="robot_name" type="string" value="$(arg robot_name)" />
    <param name="number_of_limbs" type="int" value="$(arg number_of_limbs)" />


    <!--    Define initial position of the distal plate -->
    <arg name="x" value="0.0"/>
    <arg name="y" value="0.0"/>
    <arg name="z" value="1.1"/>

    <!-- Create robot description -->
    <param name="robot_description" command="$(find xacro)/xacro '$(find cpr_physics)/launch/$(arg robot_name)/distal_plate.sdf.xacro' number_of_limbs:=$(arg number_of_limbs)"/>

    <!-- spawn model inside Gazebo -->
    <node name="spawner" pkg="gazebo_ros" type="spawn_model" output="screen" args="-sdf -model distal_plate -param robot_description -x $(arg x) -y $(arg y) -z $(arg z) -R 0  -P 0 -Y 0"/>

    <rosparam file="$(find cpr_physics)/launch/$(arg robot_name)/robot_config.yaml" command="load"/>


</launch>
